### ISSUE TITLE
azure - strategy for missing resource types

### ISSUE THREAD

**Issue by stefangordon** [cloud-custodian/cloud-custodian#6542]
TLDR; Azure supports 440,709 resource types today.  Custodian currently supports 54 Azure resources.

Custodian covers the most commonly requested Azure resources well, however there is a **very** long tail here.  

I was generating data doing things like this:

```
az provider list --expand resourceTypes/aliases > resources.json
cat resources.json | jq '[ .[].namespace + "/" + .[].resourceTypes[].resourceType , .[].resourceTypes[].aliases[]?.name ] | unique | sort' > aliases.json
```

I previously built a prototype of resource enumeration without SDK which was a relatively simple change.  I'd propose we could also consider eliminating our `resource_type` classes since they are just manual duplication from the dynamic provider data Azure already provides on every resource.   We could simply design a query for the providers we want to support (based on their capabilities array) and commit that `json` to the Azure provider to use instead of manually created resource classes.

This approach gives us several advantages:

- We can have all 440709 resources, or any subset of them, with support for generic filters and actions automatically registered based on the capabilities list provided by the Azure Provider definitions (it will tell us if it supports tagging, etc).
- We can remove SDK's for these resources unless we have legacy actions/filters for them which leverage the SDK
- We can remove a large amount of manually maintained resource metadata

I don't think we could easily generate TF or ARM templates for these resources, so automated live testing of all supported resources would be hard.  It may not be necessary since they are all being queried through standard interfaces which should not require testing for individual resource types.
